tests/naive.py

```diff
@@ -772,28 +788,41 @@ def update(self, t):
 
 
 class stumpi_egress(object):
-    def __init__(self, T, m, excl_zone=None):
+    def __init__(self, T, m, excl_zone=None, k=1):
         self._T = np.asarray(T)
         self._T = self._T.copy()
         self._T_isfinite = np.isfinite(self._T)
         self._m = m
-        if excl_zone is None:
+        self._k = k
+
+        self._excl_zone = excl_zone
+        if self._excl_zone is None:
             self._excl_zone = int(np.ceil(self._m / config.STUMPY_EXCL_ZONE_DENOM))
 
         self._l = self._T.shape[0] - m + 1
-        mp = stump(T, m)
-        self.P_ = mp[:, 0]
-        self.I_ = mp[:, 1].astype(np.int64)
-        self.left_P_ = np.full(self.P_.shape, np.inf)
-        self.left_I_ = mp[:, 2].astype(np.int64)
-        for i, j in enumerate(self.left_I_):
-            if j >= 0:
-                D = core.mass(self._T[i : i + self._m], self._T[j : j + self._m])
+        mp = stump(T, m, exclusion_zone=self._excl_zone, k=self._k)
+        self.P_ = mp[:, :k].astype(np.float64)
+        self.I_ = mp[:, k : 2 * k].astype(np.int64)
+
+        self.left_I_ = mp[:, 2 * k].astype(np.int64)
+        self.left_P_ = np.full_like(self.left_I_, np.inf, dtype=np.float64)
+
+        for i, nn_i in enumerate(self.left_I_):
+            if nn_i >= 0:
+                D = core.mass(self._T[i : i + self._m], self._T[nn_i : nn_i + self._m])
                 self.left_P_[i] = D[0]
 
         self._n_appended = 0
 
+        if self._k == 1:
```

Comment:
It feels really weird that we are flattening `self.P_` and `self.I_` when `self._k == 1` but, later in `update`, we reshape it again. This back-and-forth converting seems like a bad idea and we should fix this so that, internally, we keep things as 2D and only output 1D when we need to

Reply:
I think we can do it... but then we need to change the `tests/test_stumpi.py`. Recall that the performant version returns 1D when `k=1`. 

```

# in tests/test_stumpi.py

# here, `k` is 1.
def test_stumpi_self_join_egress():
    m = 3

    seed = np.random.randint(100000)
    np.random.seed(seed)
    n = 30
    T = np.random.rand(n)

    ref_mp = naive.stumpi_egress(T, m) 
    ref_P = ref_mp.P_.copy()
    ref_I = ref_mp.I_
    ref_left_P = ref_mp.left_P_.copy()
    ref_left_I = ref_mp.left_I_

    stream = stumpi(T, m, egress=True)

    comp_P = stream.P_.copy() # THIS IS 1D when `k=1`
    comp_I = stream.I_ # THIS IS 1D when `k=1`
    comp_left_P = stream.left_P_.copy()
    comp_left_I = stream.left_I_

    naive.replace_inf(ref_P)
    naive.replace_inf(ref_left_P)
    naive.replace_inf(comp_P)
    naive.replace_inf(comp_left_P)

    npt.assert_almost_equal(ref_P, comp_P)
    npt.assert_almost_equal(ref_I, comp_I)
    npt.assert_almost_equal(ref_left_P, comp_left_P)
    npt.assert_almost_equal(ref_left_I, comp_left_I)

```

So, we can do something similar to performant version. I mean...we can create a naiver verison that works with 2D internally, but then return 1D to user when `k=1`.

Reply:
> we can create a naiver verison that works with 2D internally, but then return 1D to user when k=1.

Yes, this is what I would expect. Would you feel comfortable doing that?

Reply:
I think so :) I will work on this soon. I will let you know if I get into trouble.